appservice/src/intTest/groovy/nu/yona/server/PinResetRequestTest.groovy

```diff
@@ -0,0 +1,153 @@
+package nu.yona.server
+
+
+import groovy.json.*
+import nu.yona.server.test.User
+
+class PinResetRequestTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Cannot verify or clear pin reset request before requesting a pin reset'()
+	{
+		given:
+		User richard = addRichard()
+
+		when:
+		User richardAfterGet = appService.getUser(appService.&assertUserGetResponseDetailsPublicDataAndVpnProfile, richard.url, true, richard.password)
+
+		then:
+		richardAfterGet.pinResetRequestUrl
+		!richardAfterGet.verifyPinResetUrl
+		!richardAfterGet.clearPinResetUrl
+
+		cleanup:
+		appService.deleteUser(richard)
+	}
+
+	def 'Request pin reset'()
+	{
+		given:
+		User richard = addRichard()
+
+		when:
+		def response = appService.yonaServer.postJson(richard.pinResetRequestUrl, [:], ["Yona-Password" : richard.password])
+
+		then:
+		response.status == 200
+		User  richardAfterGet = appService.getUser(appService.&assertUserGetResponseDetailsPublicDataAndVpnProfile, richard.url, true, richard.password)
+		!richardAfterGet.pinResetRequestUrl
+
+		// The below asserts check the path fragments. If one of these asserts fails, the Swagger spec needs to be updated too
+		richardAfterGet.verifyPinResetUrl == richard.url + "/pinResetRequest/verify"
+		richardAfterGet.clearPinResetUrl == richard.url + "/pinResetRequest/clear"
+
+		cleanup:
+		appService.deleteUser(richard)
+	}
+
+	def 'Verify pin reset confirmation code'()
+	{
+		given:
+		User richard = addRichard()
+		appService.yonaServer.postJson(richard.pinResetRequestUrl, [:], ["Yona-Password" : richard.password])
+		richard = appService.getUser(appService.&assertUserGetResponseDetailsPublicDataAndVpnProfile, richard.url, true, richard.password)
+
+		when:
+		def response = appService.yonaServer.postJson(richard.verifyPinResetUrl, """{"code" : "1234"}""", ["Yona-Password" : richard.password])
+
+		then:
+		response.status == 200
+		User  richardAfterGet = appService.getUser(appService.&assertUserGetResponseDetailsPublicDataAndVpnProfile, richard.url, true, richard.password)
+		!richardAfterGet.pinResetRequestUrl
+		richardAfterGet.verifyPinResetUrl
+		richardAfterGet.clearPinResetUrl
+
+		cleanup:
+		appService.deleteUser(richard)
+	}
+
+	def 'Clear pin reset request before verification'()
+	{
+		given:
+		User richard = addRichard()
+		appService.yonaServer.postJson(richard.pinResetRequestUrl, [:], ["Yona-Password" : richard.password])
+		richard = appService.getUser(appService.&assertUserGetResponseDetailsPublicDataAndVpnProfile, richard.url, true, richard.password)
+
+		when:
+		def response = appService.yonaServer.postJson(richard.clearPinResetUrl, [:], ["Yona-Password" : richard.password])
+
+		then:
+		response.status == 200
+		User  richardAfterGet = appService.getUser(appService.&assertUserGetResponseDetailsPublicDataAndVpnProfile, richard.url, true, richard.password)
+		richardAfterGet.pinResetRequestUrl
+		!richardAfterGet.verifyPinResetUrl
```

Comment:
Please add a line to test that the original `richard.verifyPinResetUrl` returns 400 now.